### PR TITLE
Added support for deprecated method alias

### LINKED ISSUE #4618: Deprecated checker is not able to detect deprecated aliases to existing functions/methods
### Steps to reproduce
Module `threading` contains several deprecated aliases (deprecated in python 3.7 and 3.10) of existing functions:

* threading.Thread.isAlive
* threading.currentThread
* threading.activeCount
* threading.Condition.notifyAll
* threading.Event.isSet
* threading.Thread.setName
* threading.thread.getName
* threading.Thread.isDaemon
* threading.Thread.setDaemon

Current logic of detecting deprecated methods/functions does not support aliases since it is using inferred name of function for checking against list of deprecated names. Hence, even after adding the names to list of deprecated functions, pylint produces false negative result:

Given a file `a.py`:
```python
import threading

print(threading.currentThread())
```

### Current behavior

```
$ pylint test.py

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 5.00/10, +5.00)
```

### Expected behavior

```
$ pylint test.py
************* Module test
test.py:51:6: W1505: Using deprecated method currentThread() (deprecated-method)

-------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 10.00/10, -5.00)
```

### pylint --version output

Result of `pylint --version` output:

```
$ pylint --version
pylint 2.9.0-dev1
astroid 2.6.0
Python 3.8.5 (default, Jan 27 2021, 15:41:15)
[GCC 9.3.0]
```

### PR DESCRIPTION
**Pull request by matusvalo**
## Steps

- [X] Add yourself to CONTRIBUTORS if you are a new contributor.
- [X] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [X] Write a good description on what the PR does.

## Description

This PR adds support for deprecated methods/functions which are just aliases to new names.
Examples of such methods/functions:

* threading.currentThread
* threading.activeCount
* threading.Condition.notifyAll
* threading.Event.isSet
* threading.Thread.setName
* threading.thread.getName
* threading.Thread.isDaemon
* threading.Thread.setDaemon

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|     | :sparkles: New feature |
|     | :hammer: Refactoring   |
|     | :scroll: Docs          |

## Related Issue

Closes #4618

### CODE REVIEW

File: tests/checkers/unittest_deprecated.py [hunk 1]
```diff
@@ -91,6 +91,31 @@ def deprecated_method():
...
+    def test_deprecated_method_alias(self):
+        # Tests detecting deprecated method defined as alias
+        # to existing method
+        node = astroid.extract_node(
+            """
+        class Deprecated:
+            def _deprecated_method():
+                pass
+
+            deprecated_method = _deprecated_method
```
Comment (Pierre-Sassoulas) [inline tests/checkers/unittest_deprecated.py:103]: Could you also add a functional test so it's possible to check that the warning is on the proper line ? Right now we're checking that something is raised for the whole node without checking the line and column.

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thank you for this new feature, I have a small remark, before merging but this look great 👍

File: tests/checkers/unittest_deprecated.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply tests/checkers/unittest_deprecated.py:103]: I've done it because I want your feature to be in 2.9.0 :)

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by matusvalo** [comment]
Thank you @Pierre-Sassoulas ! I come here to implement your feedback, but it seems you were faster :-D.

**Pull request merged**